src/components/AIModelsStep.tsx

```
@@ -1436,6 +1444,55 @@ export default function AIModelsStep({
+  // used to leave that model selected AND saveable: nothing between the tab
+  // and the gateway checked, so the box shipped with a primary model the
+  // credential could never route. Move the pick back to something the
+  // subscription can actually run; the picker below still SHOWS the rest,
+  // greyed, with the reason.
+  useEffect(() => {
+    if (!activeCatalog || useCustomModel || !selectedModelId) return;
+    const picked = activeCatalog.models.find((m) => m.id === selectedModelId);
+    if (!picked || isModelUsable(picked)) return;
+    const usable = activeCatalog.models.filter(isModelUsable);
+    const fallback = usable.find((m) => m.id === activeCatalog.defaultModelId) ?? usable[0];
+    if (fallback) setSelectedModelId(fallback.id);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/AIModelsStep.tsx:1470:19
  1468 |     const usable = activeCatalog.models.filter(isModelUsable);
  1469 |     const fallback = usable.find((m) => m.id === activeCatalog.defaultModelId) ?? usable[0];
> 1470 |     if (fallback) setSelectedModelId(fallback.id);
       |                   ^^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  1471 |   }, [activeCatalog, selectedModelId, useCustomModel, isModelUsable]);
  1472 |
  1473 |   // A model the SUBSCRIPTION surface does not carry is SHOWN, not hidden, and

(react-hooks/set-state-in-effect)